samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -66,7 +68,12 @@ public JobGraph plan(StreamGraph streamGraph) throws Exception {
       calculatePartitions(streamGraph, jobGraph);
     }
 
-    return jobGraph;
+    List<JobConfig> jobConfigs = jobGraph.getJobNodes().stream().map(JobNode::generateConfig).collect(Collectors.toList());
+    String planJson = new PlanJsonGenerator().toJson(jobGraph);
+    List<StreamSpec> intermediateStreams = jobGraph.getIntermediateStreams().stream()
+        .map(streamEdge -> streamEdge.getStreamSpec())
+        .collect(Collectors.toList());
+    return new StreamPlan(streamGraph, jobConfigs, intermediateStreams, planJson);
```

Comment:
It's a little weird that the JSON representation of the plan is required to construct it, and that this class needs to know about the PlanJsonGenerator. Maybe add a helper to StreamPlan that returns the JSON representation instead? Maybe even override its toString?